### PR TITLE
extend release testing

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

as we plan to keep aside branch for bug-fix releases, this branch shall be tested
the proposal is to define `release/1.*.x` branch

eventually, it would serve for any kind of releasing and in the end it just set that instead of master we test `release/1.0.x` or `release/1.1.x`

## PR review
 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.    
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ydcjeff — APPROVED** [review]

**Comment by ydcjeff** [comment]
So we need to pull request on 1.x.0 for features and 1.0.x for bug fix?

**Review by justusschock — APPROVED** [review]

**Review by tchaton — APPROVED** [review]

File: .github/workflows/pypi-release.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ jobs:
     # We do this, since failures on test.pypi aren't that bad
     - name: Publish to Test PyPI
       if: startsWith(github.event.ref, 'refs/tags') || github.event_name == 'release'
-      uses: pypa/gh-action-pypi-publish@master
+      uses: pypa/gh-action-pypi-publish@v1.4
```
Comment (ydcjeff) [inline .github/workflows/pypi-release.yml:33]: this is failing on master

Comment (SeanNaren) [reply .github/workflows/pypi-release.yml:33]: @Borda mind having a look?

Comment (Borda) [reply .github/workflows/pypi-release.yml:33]: it was working herehttps://github.com/PyTorchLightning/pytorch-lightning/actions/runs/345388960

Comment (Borda) [reply .github/workflows/pypi-release.yml:33]: fixing here https://github.com/PyTorchLightning/pytorch-lightning/pull/4516

**Pull request merged**